app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt

```diff
@@ -344,4 +391,48 @@ class BookmarksTest {
             verifyFolderTitle("3")
         }
     }
+
+    @Test
+    fun changeBookmarkParentFolderTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        browserScreen {
+            createBookmark(defaultWebPage.url)
+        }.openThreeDotMenu {
+        }.openLibrary {
+        }.openBookmarks {
+        }.openThreeDotMenu {
+        }.clickEdit {
+            verifyEditBookmarksView()
+            changeBookmarkTitle(bookmarkTitle)
+            saveEditBookmark()
+            createFolder(bookmarksFolderName)
+        }.openThreeDotMenu(bookmarkTitle) {
+        }.clickEdit {
+            clickParentFolderSelector()
+            selectFolder(bookmarksFolderName)
+            navigateUp()
+            saveEditBookmark()
+            selectFolder(bookmarksFolderName)
+            verifyBookmarkedURL(defaultWebPage.url)
+        }
```

Comment:
Is there a chance for any similar async ops we need to wait in this too?

Reply:
I added the RecyclerViewIdlingResource as you suggested. Please review it again.